internal/core/src/query/SearchBruteForce.cpp

```diff
@@ -101,14 +72,26 @@ BruteForceSearch(const dataset::SearchDataset& dataset,
             CheckRangeSearchParam(
                 config[RADIUS], config[RANGE_FILTER], dataset.metric_type);
         }
-        auto res = knowhere::BruteForce::RangeSearch(
-            base_dataset, query_dataset, config, bitset);
+        knowhere::expected<knowhere::DataSetPtr> res;
+        if (data_type == DataType::VECTOR_FLOAT) {
+            res = knowhere::BruteForce::RangeSearch<float>(
+                base_dataset, query_dataset, config, bitset);
+        } else if (data_type == DataType::VECTOR_FLOAT16) {
+            res = knowhere::BruteForce::RangeSearch<float16>(
+                base_dataset, query_dataset, config, bitset);
+        } else if (data_type == DataType::VECTOR_BFLOAT16) {
+            res = knowhere::BruteForce::RangeSearch<bfloat16>(
+                base_dataset, query_dataset, config, bitset);
+        } else if (data_type == DataType::VECTOR_BINARY) {
+            res = knowhere::BruteForce::RangeSearch<uint8_t>(
+                base_dataset, query_dataset, config, bitset);
+        }
         milvus::tracer::AddEvent("knowhere_finish_BruteForce_RangeSearch");
         if (!res.has_value()) {
             PanicInfo(KnowhereError,
-                      "failed to range search: {}: {}",
-                      KnowhereStatusString(res.error()),
-                      res.what());
+                      fmt::format("failed to range search: {}: {}",
```

Comment:
The `PanicInfo` supports to format the arguments, no need to call `fmt::format`, maybe improve this next PR

Reply:
Get.